tests/test_connection.py

```diff
@@ -391,6 +391,30 @@ def test_reconnect_gossiping(node_factory):
     l2.daemon.wait_for_log('processing now old peer gone')
 
 
+@unittest.skipIf(not DEVELOPER, "needs dev-disconnect")
+def test_reconnect_no_update(node_factory, executor):
+    """
+    This tests if the `funding_locked` is sent if we receive a
+    `channel_reestablish` message with `next_commitment_number` == 1 and
+    our `next_commitment_number` == 1.
+    """
+    disconnects = ["@WIRE_FUNDING_LOCKED"]
+    l1 = node_factory.get_node(may_reconnect=True)
+    l2 = node_factory.get_node(disconnect=disconnects, may_reconnect=True)
+
+    # For channeld reconnection
+    l1.rpc.connect(l2.info["id"], "localhost", l2.port)
+    fundchannel_exec = executor.submit(l1.fund_channel, l2, 10**6, False)
+    l1.daemon.wait_for_log(r"Retransmitting funding_locked for channel*")
+    l2.stop()
+
+    # For closingd reconnection
+    scid = fundchannel_exec.result()
+    l2.daemon.start()
+    executor.submit(l1.rpc.close, scid)
+    l1.daemon.wait_for_log(r"Retransmitting funding_locked for channel*")
```

Comment:
I also need to disconnect at `FUNDING_LOCKED` because otherwise the nodes will add updates and the test would fail. By the way, why do they add update just after funding a channel, shouldn't they keep the first commitment pair ?

I corrected the test.

Reply:
They try to change the feerate.  If you set all feerates to the same, you can suppress this, FWIW.